src/mc-efc.c

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2022-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "mc-efc-private.h"
+
+#include "mongocrypt-private.h"
+
+/* _parse_field parses and prepends one field document to efc->fields. */
+static bool
+_parse_field (mc_EncryptedFieldConfig_t *efc,
+              bson_t *field,
+              mongocrypt_status_t *status)
+{
+   bson_iter_t field_iter;
+   if (!bson_iter_init_find (&field_iter, field, "keyId")) {
+      CLIENT_ERR ("unable to find 'keyId' in 'field' document");
+      return false;
+   }
+   if (!BSON_ITER_HOLDS_BINARY (&field_iter)) {
+      CLIENT_ERR ("expected 'fields[].keyId' to be type binary, got: %d",
```

Comment:
Recommend adjusting error messages to either include index of `fields` array (e.g. `expected 'fields[0].keyid' to be type binary`) or avoid using subscript notation without an index (e.g. `expected 'keyId' to be type binary`) as `[]` alone looks like it is missing an index (format error). Suggestion applies to all other instances of `'fields[]'` in error messages in this file.

Reply:
Good point. Removed the subscript notation.